### PR TITLE
Add size limits on responses

### PR DESCRIPTION
**Pull request by jsha**
This prevents a malicious server from running the client out of memory
by returning a very large body.

Limit into_string() to 1 megabyte.
Limit individual header fields to 100 kilobytes.
Limit number of header fields to 100.

Part of #267

### CODE REVIEW

File: src/response.rs [hunk 1]
```diff
@@ -21,6 +21,10 @@ use encoding_rs::Encoding;
 pub const DEFAULT_CONTENT_TYPE: &str = "text/plain";
 pub const DEFAULT_CHARACTER_SET: &str = "utf-8";
 const INTO_STRING_LIMIT: usize = 1_024 * 1_024;
```
Comment (algesten) [inline src/response.rs:23]: 1MB sounds a bit too conservative. Could we maybe have 10 or 20?

My thinking is that ureq will is likely to be used mostly in situations that aren't super constrained. And if memory consumption is a concern, the user should consider using a streaming read anyway.

**Review by algesten — APPROVED** [review]
Looks good! One small thing.

File: src/response.rs (hunk 1 above)
Comment (jsha) [reply src/response.rs]: Yep, makes sense! I originally thought even as big as 1GB... But when writing this PR I thought "things that are UTF-8 are usually textual, and it's rare to have textual content that big." But it's easy to have a >1MB JSON file for instance.

Comment (algesten) [reply src/response.rs]: Yes that's my thinking. I have that exact thing at work. We're running AWS transcription jobs on hour long interviews. The resulting JSON files we store on S3 are big!

**Comment by algesten** [comment]
I think this is ready to merge. Looks good!

**Pull request merged**